### ISSUE TITLE
Snippet disappeared

### ISSUE THREAD

**Issue by Airblader** [moroshko/rxviz#12]
I created a snippet and saved it, giving me the link [xOvAwvo2](https://rxviz.com/v/xOvAwvo2). The snippet worked at first, verified by the fact that the person asking [this question](https://stackoverflow.com/questions/48487177/rxjs-throttletime-dynamically-change-the-throttling/48487529#48487529) was able to see the snippet, so it's also not an issue of copy-pasting the wrong link etc.

However, now (a couple minutes after creating it), the link just 404s for me. The same happened with another snippet I made in response, and that one I double-checked that the copied URL worked.

Seems like mysterious things are going on here.

**Comment by moroshko** [comment]
Here is what's going on here:

We generate the shareable link when "Visualize" is clicked (not when "Share link" is clicked). This allows us to have a one-click "copy to clipboard" functionality.

If you keep changing the code and clicking "Visualize", we would keep saving lots of snippets.
If, for example, you had 100 iterations on your code, and decide to share the last one, we ended up with unnecessary 99 snippets in the database.

This is why we decided to smartly delete unnecessary snippets from the database. But, apparently, we are not that smart :)

Once you clicked the "Share link" button, we should never delete that snippet.

I'll fix that!
